File: server/storage-service/src/main/java/com/sudoo/storageservice/controller/ImageController.java
package com.sudoo.storageservice.controller;

import com.sudo248.domain.base.BaseResponse;
import com.sudoo.storageservice.service.ImageService;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import org.springframework.web.multipart.MultipartFile;

import javax.servlet.http.HttpServletRequest;

@RestController
@RequestMapping("/images")
public class ImageController {

    private final ImageService imageService;

    public ImageController(ImageService imageService) {
        this.imageService = imageService;
    }

    @PostMapping(value = "/upload", consumes = {MediaType.MULTIPART_FORM_DATA_VALUE})
    public ResponseEntity<BaseResponse<?>> upload(
            @RequestParam(value = "image") MultipartFile image,
            @RequestParam(value = "source", required = false, defaultValue = "cloudinary") String source
    ) {
        return imageService.storeImage(image, source);
    }
}

File: server/product-service/src/main/kotlin/com/sudoo/productservice/controller/ImageController.kt
package com.sudoo.productservice.controller

import com.sudoo.domain.base.BaseController
import com.sudoo.domain.base.BaseResponse
import com.sudoo.productservice.dto.ImageDto
import com.sudoo.productservice.service.ImageService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.DeleteMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/discovery/images")
class ImageController(
    private val imageService: ImageService
) : BaseController() {

    @PostMapping
    suspend fun upsertImage(
        @RequestBody imageDto: ImageDto
    ): ResponseEntity<BaseResponse<*>> = handle {
        imageService.upsertImage(imageDto)
    }

    @DeleteMapping("/{imageId}")
    suspend fun deleteImage(
        @PathVariable("imageId") imageId: String
    ): ResponseEntity<BaseResponse<*>> = handle {
        imageService.deleteImage(imageId)
    }
}